### ISSUE TITLE
git module seems to break stderr

### ISSUE THREAD

**Issue by gnuletik** [sorin-ionescu/prezto#1869]
### Description

When enabling the git module, the `stderr` output occasionally breaks and nothing is printed on the terminal. 

### Expected behavior

When writing to `stderr`, it should be printed into the terminal.

### Steps to Reproduce

It's quite hard to reproduce this bug. I tried many different scenarios but I didn't found any simple steps to reproduce it.

```
cd some-git-repo
# write some commands to generate the issue
# e.g.
~/c/i/f/s/d/a/www ❯❯❯ e                                                     master ✭ ✱ ◼
zsh: command not found: nano
~/c/i/f/s/d/a/www ❯❯❯ e                                               ✘ 127 master ✭ ✱ ◼
zsh: command not found: nano
~/c/i/f/s/d/a/www ❯❯❯ e                                               ✘ 127 master ✭ ✱ ◼
zsh: command not found: nano
~/c/i/f/s/d/a/www ❯❯❯ e                                               ✘ 127 master ✭ ✱ ◼
zsh: command not found: nano
~/c/i/f/s/d/a/www ❯❯❯ eee                                             ✘ 127 master ✭ ✱ ◼
zsh: command not found: eee
~/c/i/f/s/d/a/www ❯❯❯ e                                               ✘ 127 master ✭ ✱ ◼
zsh: command not found: nano
~/c/i/f/s/d/a/www ❯❯❯ e                                               ✘ 127 master ✭ ✱ ◼
zsh: command not found: nano
~/c/i/f/s/d/a/www ❯❯❯ eeee                                            ✘ 127 master ✭ ✱ ◼
zsh: command not found: eeee
~/c/i/f/s/d/a/www ❯❯❯ e                                               ✘ 127 master ✭ ✱ ◼
zsh: command not found: nano
~/c/i/f/s/d/a/www ❯❯❯ e                                               ✘ 127 master ✭ ✱ ◼
zsh: command not found: nano
~/c/i/f/s/d/a/www ❯❯❯ e                                               ✘ 127 master ✭ ✱ ◼

# At some point, nothing is printed to stderr

~/c/i/f/s/d/a/www ❯❯❯ eeeee                                           ✘ 127 master ✭ ✱ ◼
~/c/i/f/s/d/a/www ❯❯❯ echo test 1>&2                                  ✘ 127 master ✭ ✱ ◼

# But stdout seems to work fine

~/c/i/f/s/d/a/www ❯❯❯ echo test                                       ✘ 127 master ✭ ✱ ◼
test
```

Sometimes, I just cd into a git repository, and the issue occurs.

I tried to disable the git module and I could not reproduce the issue.

### Versions

  - Prezto commit: 0a405494cae54e208ff550c1ba83380ed115410a
  - ZSH version: zsh 5.8
  - OS information: Archlinux (kernel 5.9.1-arch1-1)

**Comment by srijanshetty** [comment]
@gnuletik I'm not sure what is causing this issue but I know a fix for this. `nano` is the default editor for git. You should update your `$EDITOR` to either `vim` or `emacs` to fix this issues or just update `global git config` to specific a different `EDITOR`.

**Comment by gnuletik** [comment]
Hi @srijanshetty 

Thanks for the answer !
The `$EDITOR` variable was already set but setting the `$VISUAL` variable fixed the default editor for git.

However, the issue still appears when spamming other commands.

```
zsh: command not found: CD
~/c/i/f/s/d/a/www ❯❯❯ S                                                                                                                                             ✘ 127 dev ✭ ◼
zsh: command not found: S
~/c/i/f/s/d/a/www ❯❯❯ A                                                                                                                                             ✘ 127 dev ✭ ◼
zsh: command not found: A
~/c/i/f/s/d/a/www ❯❯❯ A                                                                                                                                             ✘ 127 dev ✭ ◼
zsh: command not found: A
~/c/i/f/s/d/a/www ❯❯❯ A                                                                                                                                             ✘ 127 dev ✭ ◼
zsh: command not found: A
~/c/i/f/s/d/a/www ❯❯❯                                                                                                                                               ✘ 127 dev ✭ ◼
~/c/i/f/s/d/a/www ❯❯❯ a                                                                                                                                             ✘ 127 dev ✭ ◼
zsh: command not found: a
~/c/i/f/s/d/a/www ❯❯❯ a                                                                                                                                             ✘ 127 dev ✭ ◼
zsh: command not found: a

# and now stderr is broken

~/c/i/f/s/d/a/www ❯❯❯ a                                                                                                                                             ✘ 127 dev ✭ ◼
~/c/i/f/s/d/a/www ❯❯❯                                                                                                                                               ✘ 127 dev ✭ ◼
~/c/i/f/s/d/a/www ❯❯❯                                                                                                                                               ✘ 127 dev ✭ ◼

~/c/i/f/s/d/a/www ❯❯❯ a                                                                                                                                             ✘ 127 dev ✭ ◼
~/c/i/f/s/d/a/www ❯❯❯ echo test 1>&2                                                                                                                                ✘ 130 dev ✭ ◼
~/c/i/f/s/d/a/www ❯❯❯ echo test                                                                                                                                           dev ✭ ◼
test

# stdout still works
```

**Comment by belak** [comment]
My current guess is that something in our git-info function or the external zsh-async are using stderr redirection with a command stored in a variable and that var is somehow unset sometimes.

I think we've had similar issues with people holding down Control C...

Does this seem correlated to how quickly you rerun the command, or is that not a factor?

**Comment by gnuletik** [comment]
Hi @belak,

Thanks for the answer.

Holding Ctrl-C do not reproduce the problem.

I tried this test
- cd into git repo
- press 'a'
- Wait half a second
- press enter
- Wait half a second
- Go back to the "press 'a'" step

The issue occurred after 40 occurrences.
I repeated the tests faster multiple times, sometimes it occurs after 5 occurrences, sometimes after 40.

So, it does not seems correlated to how quickly I rerun the command.

Thanks!
